core/dbt/contracts/graph/model_config.py

```diff
@@ -454,182 +440,46 @@ class TestConfig(NodeConfig):
-            schema = _validate_schema(h_cls)
-            to_validate = data
-
-        validator = jsonschema.Draft7Validator(schema)
-
-        error = jsonschema.exceptions.best_match(
-            validator.iter_errors(to_validate),
-            key=_relevance_without_strategy,
-        )
-
-        if error is not None:
-            raise ValidationError.create_from(error) from error
-
-
 @dataclass
 class EmptySnapshotConfig(NodeConfig):
     materialized: str = 'snapshot'
 
 
-@dataclass(init=False)
+@dataclass
 class SnapshotConfig(EmptySnapshotConfig):
-    unique_key: str = field(init=False, metadata=dict(init_required=True))
-    target_schema: str = field(init=False, metadata=dict(init_required=True))
+    strategy: Optional[str] = None
+    unique_key: Optional[str] = None
+    target_schema: Optional[str] = None
     target_database: Optional[str] = None
+    updated_at: Optional[str] = None
+    check_cols: Optional[Union[str, List[str]]] = None
 
-    def __init__(
-        self,
-        unique_key: str,
-        target_schema: str,
-        target_database: Optional[str] = None,
-        **kwargs
-    ) -> None:
-        self.unique_key = unique_key
-        self.target_schema = target_schema
-        self.target_database = target_database
-        # kwargs['materialized'] = materialized
-        super().__init__(**kwargs)
-
-    # type hacks...
     @classmethod
-    def _get_fields(cls) -> List[Tuple[Field, str]]:  # type: ignore
-        fields: List[Tuple[Field, str]] = []
-        for old_field, name in super()._get_fields():
-            new_field = old_field
-            # tell hologram we're really an initvar
-            if old_field.metadata and old_field.metadata.get('init_required'):
-                new_field = field(init=True, metadata=old_field.metadata)
-                new_field.name = old_field.name
-                new_field.type = old_field.type
-                new_field._field_type = old_field._field_type  # type: ignore
-            fields.append((new_field, name))
-        return fields
-
-    def finalize_and_validate(self: 'SnapshotConfig') -> SnapshotVariants:
+    def validate(cls, data):
+        super().validate(data)
+        if data.get('strategy') == 'check':
+            if not data.get('check_cols'):
+                raise ValidationError(
+                    "A 'check' snapshot must have 'check_cols'")
+            if (isinstance(data['check_cols'], str) and
+                    data['check_cols'] != 'all'):
+                raise ValidationError(
+                    f"Invalid value for 'check_cols': {data['check_cols']}")
+
+        elif data.get('strategy') == 'timestamp':
+            if not data.get('updated_at'):
+                raise ValidationError(
+                    "A 'timestamp' snapshot must have 'updated_at'")
+            if data.get('check_cols'):
+                raise ValidationError(
+                    "A 'timestamp' snapshot should not have 'check_cols'")
```

Comment:
> A snapshot configured with the timestamp strategy must not specify a check_cols strategy

Also, is this consistent with the existing behavior? I wonder if we errored in this case, or if we accepted it but ignored the config?

Reply:
It is consistent. There's a test case for it.

Reply:
Right on! I didn't call this out above, but wondering if we can also update the error message copy to be more in line with my suggestion above? I find that the more complete/precise the sentence is, the more helpful the error message is! Very open to workshopping this with you if you have alternative thoughts on the message!